components/asset-actions/buttons/BuyAssetButton.tsx

```diff
@@ -118,13 +163,26 @@ export function BuyAssetButton({
           />
         </Case>
         <Case value="buy">
-          <BuyStep
-            asset={asset}
-            order={order}
-            onValid={closeDialog}
-            quantity={quantity}
-            transactionPrice={transactionPrice}
-          />
+          {isValidationLoading ? (
+            <div className="flex items-center justify-center ">
+              <Loader size={22} className="animate-spin" />
+            </div>
+          ) : errorMessages.title ? (
```

Comment:
I don't think you need twice this ternary logic. I think it should be possible to add it before the <Case> components so you have this logic only once